crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -311,16 +311,39 @@ impl World {
         self.spawn(Observer::new(system))
     }
 
-    /// Triggers the given `event`, which will run any observers watching for it.
+    /// Triggers the given [`Event`], which will run any [`Observer`]s watching for it.
+    ///
+    /// This method consumes the event, so it cannot be used after this method is called.
```

Comment:
My `Event`s usually implement `Copy`, so that's not really true

Reply:
```suggestion
    /// This method consumes the event, so it cannot be used after this method is called,
    /// unless the event implements `Copy`.
```

Would something like this work? My motivation for this PR involves non-`Copy` event types.

Reply:
```suggestion
    /// While event types commonly implement [`Copy`], those that don't will be consumed and will no longer be accessible.
```

Alternatively

Reply:
Both are fine for me :)